.github/workflows/backstop.yml

```diff
@@ -0,0 +1,123 @@
+          drush cvapi Extension.upgrade
+          drush en civicrmtheme -y
+          drush dl bootstrap -y
+          drush en bootstrap -y
+          drush vset theme_default bootstrap
+
+      - name: Installing Shoreditch Companion Theme
+        working-directory: ${{ env.DRUPAL_THEME_DIR }}
+        run: |
+          mkdir shoreditch_companion_d7_theme
+          cd shoreditch_companion_d7_theme/
+          git clone https://github.com/compucorp/shoreditch-companion-d7-theme.git
+          drush en -y shoreditch_companion_d7_theme
+          drush vset civicrmtheme_theme_admin shoreditch_companion_d7_theme
+
+          drush sql-query "UPDATE block SET status = 0 WHERE theme='shoreditch_companion_d7_theme' AND module='civicrm' AND delta IN ('1', '2', '3', '4', '5');" -y
+          drush sql-query "UPDATE block SET status = 0 WHERE theme='shoreditch_companion_d7_theme' AND module='search' AND delta='form';" -y
+          drush sql-query "UPDATE block SET status = 0 WHERE theme='shoreditch_companion_d7_theme' AND module='system' AND delta IN ('navigation', 'powered-by');" -y
+
+          drush cc all && drush cc civicrm
+
+      - name: Create site-config.json for Backstop tests
+        working-directory: ${{ env.CIVICRM_EXTENSIONS_DIR }}/uk.co.compucorp.civicase
+        run: |
+          cd tests/backstop_data
+          touch site-config.json
+          echo "{" >> site-config.json
+          echo '"url": "http://localhost:7979",' >> site-config.json
+          echo '"drush_alias": "",' >> site-config.json
+          echo '"root": "/__w/uk.co.compucorp.civicase/uk.co.compucorp.civicase/site/web"' >> site-config.json
+          echo "}" >> site-config.json
+
+      - name: Reference Screenshots in "master" Branch
+        continue-on-error: true
+        working-directory: ${{ env.CIVICRM_EXTENSIONS_DIR }}/uk.co.compucorp.civicase/tests/backstop_data
+        run: |
+          git reset --hard
+          git checkout "${{ github.head_ref }}"
+          npm install
+          npx gulp backstopjs:setup-data
+          npx gulp backstopjs:reference
+
+      - name: Test Screenshots in "${{ github.head_ref }}" Branch
+        working-directory: ${{ env.CIVICRM_EXTENSIONS_DIR }}/uk.co.compucorp.civicase/tests/backstop_data
+        continue-on-error: true
+        run: |
+          git reset --hard
+          git checkout "${{ github.head_ref }}"
+          npx gulp backstopjs:test
+
+      - name: Prepare test report for download
+        if: ${{ always() }}
+        working-directory: ${{ env.CIVICRM_EXTENSIONS_DIR }}/uk.co.compucorp.civicase/tests/backstop_data
+        run: |
+          mkdir backstop_report
+          cp -r html_report backstop_report
+          cp -r screenshots backstop_report
+
+      - uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: backstop-report
+          path: ${{ env.CIVICRM_EXTENSIONS_DIR }}/uk.co.compucorp.civicase/tests/backstop_data/backstop_report
```

Comment:
How big are these reports on average? Do you have any strategies in mind on how to delete artifacts from old runs? The reason I'm asking is that we have a limit of 500MB of storage for these files: https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions#about-billing-for-github-actions (We're using Github Free)

Reply:
@davialexandre I will try to optimise these, but looking at the wiki
> GitHub Actions usage is free for public repositories. For private repositories, each GitHub account receives a certain amount of free minutes and storage

So it should be fine for us?

Reply:
Oh, good point! If that is really the case, then it shouldn't be a problem. I'll keep an eye on the limits anyway

Reply:
@deb1990 I've just confirmed it's not using any of the storage we have available:

![2020-07-22-154734_983x362_scrot](https://user-images.githubusercontent.com/388373/88216030-e9bb0380-cc32-11ea-9edf-6a9d7ab44875.png)